frontend/src/components/inbox/helpers/mapMessages.tsx

```diff
@@ -60,6 +61,31 @@ export function isSubsequent(
   );
 }
 
+/**
+ * @description Detects URLs in text and converts them to clickable links using Autolinker
+ */
+function formatMessageText(text: string): ReactNode {
+  try {
+    const linkedText = Autolinker.link(text, {
+      className: "chat-link",
+      newWindow: true,
+      truncate: { length: 50, location: "middle" },
+      stripPrefix: false,
+      sanitizeHtml: true,
+    });
+
+    return (
+      <span
```

Comment:
Not sure if "span" is the right element. Ideally we would adjust Message.TextContent component.